azure/azure-iot-hub-device-simulator/src/main/java/org/eclipse/examples/azure/ConfigProperties.java

```diff
@@ -32,66 +32,51 @@
     private final String proxyHost;
     private final Integer proxyPort;
     private final String proxyUsername;
     private final String proxyPassword;
 
-    private ConfigProperties(final String connectionString,
-            final String namespace,
-            final String thingId,
-            final IotHubClientProtocol connectionProtocol,
-            final String proxyHost,
-            final Integer proxyPort,
-            final String proxyUsername,
-            final String proxyPassword) {
-
-        this.connectionString = connectionString;
-        this.namespace = namespace;
-        this.thingId = thingId;
-        this.connectionProtocol = connectionProtocol;
-        this.proxyHost = proxyHost;
-        this.proxyPort = proxyPort;
-        this.proxyUsername = proxyUsername;
-        this.proxyPassword = proxyPassword;
+    private ConfigProperties(final Properties props) {
+        connectionString = getConnectionStringFromConfig(props);
+        namespace = props.getProperty("namespace");
+        thingName = props.getProperty("thing_id");
+        connectionProtocol = getConnectionProtocolFromConfig(props);
+        proxyHost = getProxyHostFromConfig(props);
+        proxyPort = getProxyPortFromConfig(props);
+        proxyUsername = props.getProperty("proxy_username");
+        proxyPassword = props.getProperty("proxy_password");
     }
 
     public static ConfigProperties getInstance() {
         if (null == instance) {
-            final Properties props = readPropertiesFromFile();
-
-            final var connectionString = getConnectionStringFromConfig(props);
-            final var namespace = props.getProperty("namespace");
-            final var thingId = props.getProperty("thing_id");
-            final var connectionProtocol = getConnectionProtocolFromConfig(props);
-            final var proxyHost = getProxyHostFromConfig(props);
-            final var proxyPort = getProxyPortFromConfig(props);
-            final var proxyUsername = props.getProperty("proxy_username");
-            final var proxyPassword = props.getProperty("proxy_password");
-
-            instance =
-                    new ConfigProperties(connectionString, namespace, thingId, connectionProtocol, proxyHost, proxyPort,
-                            proxyUsername, proxyPassword);
+            final Properties props = tryToReadPropertiesFromFile();
+
+            instance = new ConfigProperties(props);
             LOGGER.debug("Loaded config properties.");
         }
         return instance;
     }
 
-    private static Properties readPropertiesFromFile() {
-        final Properties props = new Properties(System.getProperties());
-
-        try (final InputStream in = ConfigProperties.class.getClassLoader()
-                .getResourceAsStream(CONFIG_PROPERTIES_FILE)) {
-            props.load(in);
-        } catch (final IOException ioe) {
+    private static Properties tryToReadPropertiesFromFile() {
```

Comment:
I prefer to supply the file name as argument here. Then the reference to the constant just has to be used by the caller of this method (and throwing the exception then probably fits in one line).